### PR TITLE
Add BuiltinVars to the GoldenTest YAML model

### PR DESCRIPTION
**Pull request by drevell**
- Creates a new version of the GoldenTest model in v1beta3 following
  templates/model/README.md.
- Add the new BuiltinVars field to the new v1beta3 goldentest model.
  This will be used to implement #362.
- Rename InputValue to VarValue; it used to be the case that all vars
  are template inputs, but that stopped being true after the
  introduction of builtin vars like `_git_tag.`

### CODE REVIEW

File: templates/commands/goldentest/test_funcs.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ import (
 	"github.com/abcxyz/abc/templates/common"
 	"github.com/abcxyz/abc/templates/common/render"
 	"github.com/abcxyz/abc/templates/model/decode"
-	goldentest "github.com/abcxyz/abc/templates/model/goldentest/v1alpha1"
+	goldentest "github.com/abcxyz/abc/templates/model/goldentest/v1beta3"
```
Comment (capri-xiyue) [inline templates/commands/goldentest/test_funcs.go:31]: Curious to learn the version management in abc repo.  Looks like in `templates/model/goldentest`, we maintain different versions of goldentest for example v1alpha1 and v1beta3. But in `templates/commands/goldentest`, we use the latest version which is v1beta3. In that case, who will be the user of `templates/model/goldentest/v1alpha1`, I guess the users with older cli versions? Does the command `cli` has its own version?

File: templates/model/decode/decode_test.go [hunk 2]
```diff
@@ -150,7 +151,7 @@ inputs:
   - name: 'foo'
```
Comment (capri-xiyue) [inline templates/model/decode/decode_test.go:151]: Do we need to add `builtin_vars` in the test case to verify the latest goldentest works?

**Comment by capri-xiyue** [comment]
Curious to learn the difference between BuiltinVars and Input. Is my understanding correct?
1. Input is defined the template spec but BuiltinVars is not?
2. Do we need to have any validation for BuiltinVars similar to Input? Looks like we don't have any specific validation besides non-zero.

In addition, curious to learn _git_*  vars CUJ in goldentest?

File: templates/commands/goldentest/test_funcs.go (hunk 1 above)
Comment (drevell) [reply templates/commands/goldentest/test_funcs.go:31]: The docs about api_version changes are a little rudimentary, but they're in templates/model/README.md (this was also linked in the PR description)

The fundamental design is that old `api_version`s are supported, they're just auto-upgraded to the latest api_version. The abc CLI app logic is written to use only the latest api_version. See the upgrade.go files and the `DecodeValidateUpgrade()` function to understand how api_version upgrading works.

File: templates/model/decode/decode_test.go (hunk 2 above)
Comment (drevell) [reply templates/model/decode/decode_test.go:151]: Yes, there's much more coming in future PRs.

**Comment by drevell** [comment]
> 1. Input is defined the template spec but BuiltinVars is not?

Right.

> 2. Do we need to have any validation for BuiltinVars similar to Input? Looks like we don't have any specific validation besides non-zero.

Yes, that will be a future PR.

> In addition, curious to learn _git_* vars CUJ in goldentest?

I just sent you the internal design doc over chat.

> I'm thinking how this BuiltInVars would change the interface of #316? Like `abc templates golden-test new-test. Example: abc templates golden-test new-test --location . --test-name my-new-test --input service_account=abc123@foo.com --input somethingelse=bar --builtin_vars gitkey1=gitvalue1 --builtin_vars gitkey2=gitvalue2`, and the `--prompt` is just for input but NOT for builtin_vars as builtinVars is not defined in template?

I didn't think about, good question. Your suggestion sounds good to me. But I'd prefer `--builtin-var` (singular, no underscore) rather than (`--builtin_vars`) for consistency with the other flag names.

File: templates/commands/goldentest/test_funcs.go (hunk 1 above)
Comment (capri-xiyue) [reply templates/commands/goldentest/test_funcs.go:31]: Thanks for explanation!

**Review by capri-xiyue — APPROVED** [review]

**Pull request merged**